tests/test_decorator.py

```diff
@@ -0,0 +1,47 @@
+from types import SimpleNamespace
+from unittest.mock import patch
+
+import pytest
+
+from woodchipper.decorator import arg_logger, missing, pluck_value
+
+
+def test_arg_logger_invokes_logging_context_with_arguments():
+    @arg_logger(Foo="foo", Bar="bar", NestedObjID="key.nested_obj.id", NestValue="nest.value")
+    def foo(bar, key=None, foo="hello", **kwargs):
+        pass
+
+    with patch("woodchipper.decorator.LoggingContext", autospec=True) as mocked:
+        foo("bar value", key={"nested_obj": {"id": "some_id"}})
+
+    assert mocked.called
+    assert not mocked.call_args.kwargs
+    assert mocked.call_args.args == (
+        {"Bar": "bar value", "NestedObjID": "some_id", "NestValue": missing, "Foo": missing},
```

Comment:
Do we really want to pass the singleton through to the LoggingContext? And if so, do we need to add additional logic there to handle values that are instances of the singleton? Because at present, this violates the type contract for LoggableValue.